### DISCUSSION TITLE
Anyway to detect specific device disconnect?

### DISCUSSION THREAD

**Discussion by kainbryanjones** [yume-chan/ya-webadb#548] [Q&A] [answered]
We're using ya-webadb at the company I'm working for to develop a tool for testing our app. Currently all runs well, but I can't get the UI to update when the device disconnects.

WebUSB supports the ondisconnect function which passes the serialNumber through the event object. I was planning on using this to detect if the device using Adb had disconnected but the Adb object has no 'serialNumber' parameter. Is there anything I'm missing here where I can just add a listener?

Using React just want to update the state.

Thanks.

**Comment by yume-chan** [comment, marked as answer]
In a previous version `Adb` class uses `AdbBackend` instances directly to communicate with devices, so it was possible to read the serial number via `adb.backend.serial`. But now `Adb` and `AdbBackend` are decoupled, `Adb` only requires a `ReadableWriablePair` and `AdbBackend` has a `connect` method to provide that, it's not possible to read the serial number from `Adb` instance.

You need to keep the backend instance or at least the serial number somewhere else in your code, like the demo project does:

https://github.com/yume-chan/ya-webadb/blob/26fe783405ebb0e44e6ef3f004bcf242c7197961/apps/demo/src/state/global.ts#L16-L18

https://github.com/yume-chan/ya-webadb/blob/26fe783405ebb0e44e6ef3f004bcf242c7197961/apps/demo/src/state/global.ts#L32-L35

Having a `serial` on `Adb` instances does make some sense. I will add it in next version.

**Reply by kainbryanjones** [reply]
Awesome! Thanks for the speedy reply :) Keep up the good work, this library is great!
